src/jsdoc_transformer.ts

```diff
@@ -542,6 +542,24 @@ export function jsdocTransformer(
         return memberDecl ? [decl, memberDecl] : [decl];
       }
 
+      /** Returns the `this` type in this context, or undefined if none. */
+      function getContextThisType(node: ts.Node): ts.Type|undefined {
+        do {
+          if (ts.isClassDeclaration(node) && node.name) {
+            return typeChecker.getTypeAtLocation(node.name);
+          }
+          if (ts.isFunctionDeclaration(node) && node.parameters.length > 0) {
+            const firstParam = node.parameters[0];
+            if (ts.isIdentifier(firstParam.name) &&
+                transformerUtil.getIdentifierText(firstParam.name) === 'this' && firstParam.type) {
+              return typeChecker.getTypeAtLocation(firstParam.type);
+            }
+          }
+          node = node.parent;
+        } while (node);
```

Comment:
I'm surprised this compiles, because Node isn't declared as nullable.

...I bet node.parent isn't declared as nullable either. :(

Reply:
Yes. Changed the declaration to `|undefined` and also converted to a while loop.